apps/src/templates/sectionProgress/standards/standardsConstants.js

```diff
@@ -1 +1,6 @@
 export const cstaStandardsURL = 'https://www.csteachers.org/page/standards';
+
+export const TeacherScores = {
+  COMPLETE: 100,
+  INCOMPLETE: 0
```

Comment:
Should it be TestResults.GENERIC_FAIL for INCOMPLETE and TestResults.ALL_PASS for COMPLETE?

Reply:
sorry, yes, `TestResults` is correct!

I'm still not convinced whether `GENERIC_FAIL` is the correct value for someone with no progress. it could also be `NO_TESTS_RUN`. If the idea is to merge teacher scores with user progress, then I would suggest looking to see how the code handles different values (or no values) to help inform a decision.

Reply:
I think it depends on the level type. I ran into something similar when trying to figure out if a level has not been started.

Reply:
I'm thinking about `UserLevel.best_result` and `TeacherScore.score` as a completely different set of values, which will never be directly compared with one another or to the same "threshold" to determine completeness.  A teacher_score of 100 means that the teacher has indicated they think the student completed the level; 0 means they the teacher has indicated they think the student has not completed the level. 

The important part for this feature and the future is knowing which to look at when. In the case of standards we'll only look at teacher_score, not best_result, for unplugged and only look at best_result, not teacher_score, for plugged levels. 

In the future I can imagine something like this (roughly, obviously 😄) to check progress: 

def is_complete?(studentX, levelY)
  user_levelXY = UserLevel.find(level_id: levelY, user_id: studentX)
  if user_levelXY 
     teacher_scoreXY = TeacherScore.where(user_level_id: user_levelXY.id).last 
     if teacher_scoreXY 
        teacher_score_more_recent? = teacher_scoreXY.timestamp > user_levelXY.updated_at
        is_complete =   teacher_score_more_recent ?
           TeacherScoreXY.score >= TeacherScores.COMPLETE :
            UserLevelXY.best_result >= TestResults.ALL_PASS
      else 
          is_complete = UserLevelXY.best_result >= TestResults.ALL_PASS
     end 
  end
  is_complete
end 

Example: 

1.) Student X "completes" level Y based on our current progress calculation -> `UserLevelXY.best_result = TestResults.ALL_PASS` timestamp 10am 

2.) Teacher "resets" level Y for student X -> `TeacherScoreXY.score = TeacherScores.INCOMPLETE` timestamp 11am 

=> false 

Do you think there's an advantage for the best_result field and teacher_score to have aligned possible values?

Reply:
I guess I had made some assumptions related to this: 

1. when a teacher marks a student as having completed a level, the bubble should show up as green for the student as well as the teacher
2. progress resetting will work similarly under-the-hood to marking unplugged levels with teacher scores

when we send progress down to the client, we are currently just sending a batch of best_result scores. the idea I was proposing was to merge the teacher scores into this data, so that the client still just sees a bunch of best_result scores. sending the teacher scores down separately, and then doing the score merging logic on the client, would work too, but it seems like the more complicated solution to me.

adding to the complexity, once we implement progress resetting, we don't just need to know whether a level is_complete, we also want to show that a bubble appears empty (not started) if progress has been reset on that level. to do that, we'd need another client-side check against special thresholds.

generally, it seems simpler to me to use the same scale and merge the values, rather than passing both values and their special thresholds around and comparing them when you need them. I'd rather only know about the difference between teacher_score and user_level if we are debugging lost progress.

tomorrow, let:
1. get on the same page about whether the assumptions I listed earlier are valid, and
2. think through how teacher score will be used in all the parts of the product it will affect, not just the standards report.

Reply:
Uh oh - is this the more complicated solution you're suggesting I avoid? https://github.com/code-dot-org/code-dot-org/pull/33341/files

Reply:
per top-level discussion, the constants as they are currently defined look good! Please add a comment describing how these differ from `TestResults` before merging.